### PR TITLE
fixed bug which made `scalacAllOptions` unusable / updated dependencies

### PR DESCRIPTION
**Pull request by kailuowang**

### CODE REVIEW

File: src/main/scala/sbtcatalysts/CatalystsPlugin.scala [hunk 1]
```diff
@@ -253,7 +253,12 @@ trait CatalystsBase {
   )
 
   /** Combines all scalac options.*/
-  lazy val scalacAllOptions = scalacCommonOptions ++ scalacLanguageOptions ++ scalacStrictOptions ++ partialUnificatonSettings
+  lazy val scalacAllOptions = scalacCommonOptions ++ scalacLanguageOptions ++ scalacStrictOptions
```
Comment (andyscott) [inline src/main/scala/sbtcatalysts/CatalystsPlugin.scala]: Let's add types to these declarations so we can't accidentally do it again.

File: src/main/scala/sbtcatalysts/CatalystsPlugin.scala [hunk 2]
```diff
@@ -253,7 +253,12 @@ trait CatalystsBase {
...
+
+  /** all scalac options as a settings including the partialUnification and xlint **/
+  lazy val scalacAllSettings = Seq(
```
Comment (andyscott) [inline src/main/scala/sbtcatalysts/CatalystsPlugin.scala]: `lazy val scalacAllSettings: Seq[Setting[_]] = ...` works here, I think

**Comment by andyscott** [comment]
Note: I wanted to just leave those as comments instead of approving or rejecting the PR, but I can't sort out how to revert my rejection-- so I changed it to an approval since we're good to go either way.

File: src/main/scala/sbtcatalysts/CatalystsPlugin.scala (hunk 1 above)
Comment (kailuowang) [reply src/main/scala/sbtcatalysts/CatalystsPlugin.scala]: good point. added

**Comment by kailuowang** [comment]
Thanks @andyscott I added the return type signatures.

**Comment by andyscott** [comment]
Thanks! Merge away :).

And thank you for dedicating your time to an impressive amount of work on the Typelevel ecosystem!

**Pull request merged**